application/app.py

```diff
@@ -1,297 +1,294 @@
-                players=player_count
-            )
-
-    except Exception as e:
-        print(e)
-        return jsonify(message="There was an error"), 500
-
-@app.route("/api/game/start", methods=["GET"])
-@requires_auth
-def begin_game():
-    """This endpoint is for once a player is ready to start. Will begin a game if all players (3+)
-    are ready or there are 7 players
-    Inputs - player_id
-    Outputs -
-        Game not started - status comment
-        Game started - status comment, game (player, cards)"""
-    player = get_player(request.args.get('player_id'))
-
-    # Only continues if player was not already set as ready
-    if player.ready is False:
-        player.ready = True
-    else:
-        print(player.id, "was already set as ready")
-        return game_status()
-
-    db_committing_function(player)
-
-    ############################
-    # Checks if game can begin #
-    ############################
-    players = get_players(player=player)
-    player_count = len(players)
-
-    if player_count > 2:
-        if player_count < 7:
-            for p in players:
-                if p.ready is False:
-                    print("Player "+ str(p.id) +" not ready")
-                    return jsonify(status="Waiting")
-
-        # Game can begin
-        game = get_game(player=player)
-        game.started = True
-
-        # Sets up neighbours and first round
-        set_player_neighbours(players)
-        deal_wonders(players)
-        age_calcs_and_dealing(players, game)
-
-        # DB update and begin game
-        db_committing_function(game)
-        return jsonify(
-            status="Starting",
-            game=print_json(player, players=players)
-        )
-
-    else:
-        print("not enough players")
-        return jsonify(status="Waiting")
-
-@app.route("/api/game/play_card", methods=["GET"])
-@requires_auth
-def play_card():
-    """Endpoint for all card playing actions
-    Inputs - player_id, card_id, discarded (bool), and for_wonder(bool)
-    Outputs - status comment"""
-    player = get_player(request.args.get('player_id'))
-    card = get_card(request.args.get('card_id'))
-    discarded = false_true(request.args.get('discarded'))
-    for_wonder = false_true(request.args.get('for_wonder'))
-
-    if process_card(card, player, discarded, for_wonder):
-        return jsonify(status="Card played")
-    else:
-        return jsonify(
-            status="Invalid move",
-            game=print_json(player)
-        )
-
-@app.route("/api/game/end", methods=["GET"])
-@requires_auth
-def end_game():
-    """Endpoint for ending a game mid game
-    Inputs - player_id
-    Outputs - status comment"""
-    player = get_player(request.args.get('player_id'))
-    game = get_game(player=player)
-    game.complete = True
-    db.session.add(game)
-
-    try:
-        print('Ending game')
-        db.session.commit()
-        return jsonify(message="Success")
-
-    except Exception as e:

```

Comment:
This looks like it's deleting the catch block

Reply:
Good spotting